### PR TITLE
Create TeacherDetails, PresentationGroup and RequestTableGroupAndMentor Models

### PR DESCRIPTION
**Pull request by kunaladarsh**

### CODE REVIEW

File: models/TeacherDetailModel.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+const validator = require('validator');
+const bcrypt = require('bcryptjs');
+
+const teacherDetailSchema = new mongoose.Schema({
+    tid: {
+        type: String,
+        require: [true, 'Please enter your Id'],
+        unique: true,
+        trim: true,
+        lowercase: true
```
Comment (ShivaShukla08) [inline models/TeacherDetailModel.js:12]: Hi Adarsh, I think tid will be consisting of digits only. So there is no need of lowercase.

File: models/TeacherDetailModel.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    },
+
+    age: {
+        type: Number,
+        require: true,
+        max: [100, 'Age are not allowed greater than 100']
+    },
+
+    gender: {
+        type: String,
```
Comment (ShivaShukla08) [inline models/TeacherDetailModel.js:27]: Here give your own set of allowed gender categories, so that no one can enter some random string.

Comment (kunaladarsh) [reply models/TeacherDetailModel.js:27]: okay

File: models/TeacherDetailModel.js [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        require: true
+    },
+
+    personalMail: {
+        type: String,
```
Comment (ShivaShukla08) [inline models/TeacherDetailModel.js:31]: Add validator

**Review by ShivaShukla08 — APPROVED** [review]
Great! Work Adarsh

**Pull request merged**